File: competitive-programming/src/edu/algos/li01_math/L1304_FindNUniqueIntegersSumUpToZero.java
package edu.algos.li01_math;

import java.util.Arrays;

public class L1304_FindNUniqueIntegersSumUpToZero {

	public static void main(String[] args) {
		L1304_FindNUniqueIntegersSumUpToZero obj = new L1304_FindNUniqueIntegersSumUpToZero();
		
		System.out.println(Arrays.toString(obj.sumZero(5)));
	}

	/*
	 * Problem: https://leetcode.com/problems/find-n-unique-integers-sum-up-to-zero/
	 * Idea: Just 1,2,.. -2,-1 from both sides of array
	 * Companies: Microsoft
	 * 
	 * 
	 * */
	public int[] sumZero(int n) {
        int[] result = new int[n];
        for(int i=0; i<n/2; i++){
            result[i] = i+1;
            result[n-1-i] = -(i+1);
        }
        
        return result;
    }
}

File: dsa/concepts/20_math/python/l1304_find_nunique_integers_sum_up_to_zero.py
# Problem: https://leetcode.com/problems/find-n-unique-integers-sum-up-to-zero/
# Idea: Just 1,2,.. -2,-1 from both sides of array
# Companies: Microsoft
def sum_zero(n):
    result = [0] * n

    for i in range(n // 2):
        result[i] = i + 1
        result[n - 1 - i] = -(i + 1)

    return result


# ------------------ Test (Same as Java main) ------------------

print(sum_zero(5))  # Example output: [1, 2, 0, -2, -1]
